### ISSUE TITLE
Non-identical inherited components in Fluid.Fittings

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#627]
**Reported by per&#x2e;ostlund on 12 Oct 2011 13:58 UTC**
The following models:
```
Modelica.Fluid.Fittings.SimpleGenericOrifice
Modelica.Fluid.Fittings.BaseClasses.BaseModel
Modelica.Fluid.Fittings.BaseClasses.BaseModelNonconstantCrossSectionArea
```
extends both Modelica.Fluid.Interfaces.PartialTwoPortTransport and Modelica.Fluid.Interfaces.PartialLumpedFlow. Both of those models contain a m_flow component, but the components are not identical since they have different modifiers (PartialLumpedFlow modifies stateSelect of m_flow, PartialTwoPortTransport doesn't). This issue affects both MSL 3.1 and trunk, and presumably all versions between.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/627

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 16 Nov 2012 05:08 UTC**
This bug is also in the /trunk.
It could be easily fixed (i think??) 
by replacing the definition in PartialTwoPortTransport with the one in PartialLumpedFlow:
```
// Variables defined by the flow model
        Medium.MassFlowRate m_flow(
           min=if allowFlowReversal then -Modelica.Constants.inf else 0,
           start = m_flow_start,
           stateSelect = if momentumDynamics == Types.Dynamics.SteadyState then StateSelect.default else
                                     StateSelect.prefer)
      "mass flow rates between states";

        // Parameters
        parameter Modelica.Fluid.Types.Dynamics momentumDynamics=system.momentumDynamics
      "Formulation of momentum balance"
          annotation(Dialog(tab="Assumptions", group="Dynamics"), Evaluate=true);
```

**Comment by modelica-trac-importer** [comment]
**Changelog removed by adrpo on 16 Nov 2012 05:08 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by fcasella on 3 Dec 2012 14:37 UTC**
Fixed in /trunk, 58d27a8747a415c26fd626f6a5c4d4dc7e18acf5: added extra modifiers to the extend clauses, in order to make inherited components identical.
